[17.0][MIG] purchase_variant_configurator: Migration to 17.0

BhaveshHeliconia opened the pull request.

sbidoul committed:
[ADD] purchase_variant_configurator

OCA Transbot updated translations from Transifex

chienandalu committed:
[MIG] purchase_variant_configurator: Migration to 10.0

pedrobaeza committed:
[FIX] *_variant_configurator*: Adjust things for proper purchase on confirm

OCA Transbot updated translations from Transifex

[UPD] Update purchase_variant_configurator.pot

ernestotejeda committed:
[MIG] purchase_variant_configurator: Migration to 11.0

pedrobaeza committed:
[FIX] *_variant_configurator: don't lose attribute on save

cubells committed:
[FIX] sale_variant_configurator: attribute lines are not created

SalahAdDin committed:
[11.0][FIX] Filter products according to the view

- saleable products for Sale Orders.
- purchable products for Purchase Orders.

[ADD] icon.png

[UPD] README.rst

[UPD] Update purchase_variant_configurator.pot

ThomasBinsfeld committed:
[MIG] purchase_variant_configurator: pre-commit

ThomasBinsfeld committed:
[MIG] purchase_variant_configurator from 11.0 to 13.0

[UPD] Update purchase_variant_configurator.pot

[UPD] README.rst

ibuioli committed:
Added translation using Weblate (Spanish (Argentina))

Translated using Weblate (Spanish (Argentina))

Currently translated at 100.0% (1 of 1 strings)

Translation: product-variant-13.0/product-variant-13.0-purchase_variant_configurator
Translate-URL: https://translation.odoo-community.org/projects/product-variant-13-0/product-variant-13-0-purchase_variant_configurator/es_AR/

francesco-ooops committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (1 of 1 strings)

Translation: product-variant-13.0/product-variant-13.0-purchase_variant_configurator
Translate-URL: https://translation.odoo-community.org/projects/product-variant-13-0/product-variant-13-0-purchase_variant_configurator/it/

[UPD] README.rst

alexbottemanne committed:
[MIG] purchase_variant_configurator: Migration to 16.0

Co-authored-by: Alexandra Bottemanne

TT46596

Commit:
[UPD] Update purchase_variant_configurator.pot

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-variant-16.0/product-variant-16.0-purchase_variant_configurator
Translate-URL: https://translation.odoo-community.org/projects/product-variant-16-0/product-variant-16-0-purchase_variant_configurator/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (9 of 9 strings)

Translation: product-variant-16.0/product-variant-16.0-purchase_variant_configurator
Translate-URL: https://translation.odoo-community.org/projects/product-variant-16-0/product-variant-16-0-purchase_variant_configurator/it/

Commit:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (9 of 9 strings)

Translation: product-variant-16.0/product-variant-16.0-purchase_variant_configurator
Translate-URL: https://translation.odoo-community.org/projects/product-variant-16-0/product-variant-16-0-purchase_variant_configurator/pt_BR/

victoralmau committed:
[MIG] purchase_variant_configurator_on_confirm: Merged into purchase_variant_configurator and migration to 16.0

TT46597

Commit:
[UPD] Update purchase_variant_configurator.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: product-variant-16.0/product-variant-16.0-purchase_variant_configurator
Translate-URL: https://translation.odoo-community.org/projects/product-variant-16-0/product-variant-16-0-purchase_variant_configurator/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (20 of 20 strings)

Translation: product-variant-16.0/product-variant-16.0-purchase_variant_configurator
Translate-URL: https://translation.odoo-community.org/projects/product-variant-16-0/product-variant-16-0-purchase_variant_configurator/it/

SottomaiorMacedoTec committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 100.0% (20 of 20 strings)

Translation: product-variant-16.0/product-variant-16.0-purchase_variant_configurator
Translate-URL: https://translation.odoo-community.org/projects/product-variant-16-0/product-variant-16-0-purchase_variant_configurator/pt_BR/

BhaveshHeliconia committed:
[IMP] purchase_variant_configurator: pre-commit auto fixes

victoralmau committed:
[FIX] purchase_variant_configurator: Unit of measure in product templates

Steps to reproduce the error:
- Adds the 'Manage Multiple Units of Measure' permission to the user.
- Go to a quotation and add a line indicating a Product Template with several attributes (Conference Chair for example).
- You must be able to select a unit of measure.

TT55224

victoralmau commented:
Please, cherry-pick https://github.com/OCA/product-variant/pull/396 to commit history (before migration).

BhaveshHeliconia commented:
@victoralmau, Please Review.

pedrobaeza committed:
[FIX] puchase_variant_configurator: Avoid error when similar sellers found

The standard code for selecting the seller (product.supplierinfo)
performs a sorted over sequence, min_qty, price and ending with id.
When calling it from this module with a virtual record, as we do not
having yet the variant created, .id gets NewId type records (even if
the supplierinfo exists), so we get the error:

TypeError: '<' not supported between instances of 'NewId' and 'NewId'

For avoiding this, let's remove the id from the sorting criteria as
a workaround.

TT55224

victoralmau commented:
Please, cherry-pick https://github.com/OCA/product-variant/pull/399 to commit history (before migration).

BhaveshHeliconia committed:
[MIG] purchase_variant_configurator: Migration to 17.0

BhaveshHeliconia commented:
@victoralmau, Please Review.

victoralmau commented on purchase_variant_configurator/tests/test_product_product.py:
```diff
@@ -0,0 +1,83 @@
+        self.assertEqual(
+            self.product_pricelist.currency_id,
+            currency,
+            "The pricelist should use the USD currency.",
+        )
+
+    def test_pricelist_price_rule(self):
+        """Test adding a price rule to the pricelist."""
+        product = self.env["product.product"].create(
+            {
+                "name": "Test Product",
+            }
+        )
+        price_rule = self.env["product.pricelist.item"].create(
+            {
+                "pricelist_id": self.product_pricelist.id,
+                "applied_on": "0_product_variant",
+                "product_id": product.id,
+                "fixed_price": 10.0,
+            }
+        )
+        self.assertIn(
+            price_rule,
+            self.product_pricelist.item_ids,
+            "The price rule should be added to the pricelist.",
+        )
+        self.assertEqual(
+            price_rule.fixed_price,
+            10.0,
+            "The fixed price for the price rule should be 10.0.",
+        )
+
+    def test_pricelist_deletion(self):
+        """Test that the pricelist can be deleted without issues."""
+        pricelist_id = self.product_pricelist.id
+        self.product_pricelist.unlink()
+        deleted_pricelist = self.env["product.pricelist"].browse(pricelist_id)
+        self.assertFalse(
+            deleted_pricelist.exists(), "The pricelist should be deleted successfully."
+        )
```
I think this is not necessary to test it because this module does not change anything in the rates, right?

victoralmau commented on purchase_variant_configurator/views/inherited_purchase_order_views.xml:
```diff
@@ -0,0 +1,105 @@
+                             state field is a related one that is not filled until saved -->
+                    <field
+                        name="product_tmpl_id"
+                        attrs="{'invisible': [('display_type', '!=', False)], 'readonly': [('state', 'in', ('purchase', 'to approve','done', 'cancel'))]}"
```
According to the migration guide you must change the `attrs` to invisible (the same in the rest of the fields with “_similar_” changes).

victoralmau commented on purchase_variant_configurator/views/inherited_purchase_order_views.xml:
```diff
@@ -0,0 +1,105 @@
+                        colspan="2"
+                    >
+                        <tree create="0" delete="0" editable="1">
+                            <field name="owner_model" invisible="1" />
```
```suggestion
                            <field name="owner_model" column_invisible="True" />
```
Similar in other fields to avoid displaying them.
![ejemplo](https://github.com/user-attachments/assets/d46bce3a-6881-4816-b7de-6c28fcc0c184)

BhaveshHeliconia commented:
@victoralmau, Thank you for your suggestions! I have made all the changes as per the comments above. Could you please review?

victoralmau commented on purchase_variant_configurator/views/inherited_purchase_order_views.xml:
```diff
@@ -0,0 +1,103 @@
+                            />
+                        </tree>
+                    </field>
+                    <field name="can_create_product" column_invisible="True" />
```
```suggestion
                    <field name="can_create_product" invisible="True" />
```

victoralmau commented on purchase_variant_configurator/views/inherited_purchase_order_views.xml:
```diff
@@ -0,0 +1,103 @@
+                expr="//field[@name='order_line']/form//field[@name='product_id']"
+                position="after"
+            >
+                <field name="product_id_configurator_domain" column_invisible="True" />
```
```suggestion
                <field name="product_id_configurator_domain" invisible="True" />
```

BhaveshHeliconia commented:
@victoralmau, It's Fixed. Please Review.

pedrobaeza commented:
/ocabot migration purchase_variant_configurator

pedrobaeza reviewed (approved):
/ocabot merge nobump
